src/Validator/Constraints/NoParticipationDuplicateValidator.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Starfleet Project.
+ *
+ * (c) Starfleet <msantostefano@jolicode.com>
+ *
+ * For the full copyright and license information,
+ * please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace App\Validator\Constraints;
+
+use App\Entity\Conference;
+use App\Entity\User;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+class NoParticipationDuplicateValidator extends ConstraintValidator
+{
+    public function __construct(
+        private Security $security
+    ) {
+    }
+
+    /** @var Conference */
+    public function validate($conference, Constraint $constraint): void
+    {
+        if (!$conference instanceof Conference) {
+            return;
+        }
+
+        if (!$this->security->isGranted('IS_AUTHENTICATED_FULLY')) {
+            throw new AccessDeniedException();
+        }
```

Comment:
This code does not belong to a validator.

If you want to be sure there is a use, you can use something like

```suggestion
        $user = $this->security->getUser() ?? throw new \LogicException('A user must exist');
```